Added security answers and questions

Akseluhr opened the pull request:
Hello again @bregman-arie !

We added a few answers to already existing questions. Additionally, we created questions about software supply chain, package management and security concerns related to the topics. We are more than happy to hear your feedback and fix potential issues if there are any, and kudos to you and the other contributors in this repo!

Best regards,
@Akseluhr & @Abdullah1428

Akseluhr committed:
Added questions and answers

Added 15 questions (including answers) and 10 answers to existing questions.

Akseluhr committed:
Fixed links/formatation

Akseluhr committed:
fixed another typo

Akseluhr committed:
Fixed typo

Akseluhr committed:
Fixed typo

bregman-arie commented on exercises/security/README.md:
```diff
@@ -423,3 +478,133 @@ Practically this means that a single IP can server multiple web services/pages,
+</b></details>
+
+<details>
+<summary> Briefly suggest some solutions related to those threats from a developer perspective. 
```
If this is a follow up to another question (let's say the one above) I would explicitly rephrase it as "Following the examples for security threats on supply chain, briefly suggest some solutions related to those threats from a developer perspective"

bregman-arie commented on exercises/security/README.md:
```diff
@@ -423,3 +478,133 @@ Practically this means that a single IP can server multiple web services/pages,
+
+  * Carefully investigate the third party vendor’s security practices
+  * Conduct an incident response plan that should be followed if an attack occurs. 
+  * Raise awareness among employees of potential threats and critical attributes of the product and/or organization. 
```
It doesn't sound like something a developer would do, to be honest.

bregman-arie commented on exercises/security/README.md:
```diff
@@ -423,3 +478,133 @@ Practically this means that a single IP can server multiple web services/pages,
+  * Raise awareness among employees of potential threats and critical attributes of the product and/or organization. 
+</b></details>
+
+<details>
```
This feels like a commercial :) Let's remove this one

bregman-arie replied:
Same comment

bregman-arie reviewed (changes requested):
First of all, thank you both for the effort. You added quite a lot of questions and answers and personally, I really appreciate it and I'm sure the community appreciates it as well.

I do want to provide some constructive feedback, that you can use for future contributions, whether it's this project or another one:

1. Keep pull requests small. It took me quite a lot of time to review this PR. If you want to make sure that your contributions are easily reviewed in the future and you actually get attention from different reviewers, you have to try and keep them small and focused on one topic/issue/task.

2. Be consistent with the format: Some answers include a blank line under the question, and some don't. Some answers mention the source while some don't. It makes sense for an answer to not mention a source if it was written by you, but in this case I see answers from different sources, but without reference to these sources.

3. Be mindful of the content. It's easy to ask a question and copy-paste the answer from some site but always question the answer. See if it makes sense to you If you like it if you think it's something that will be clear to people or perhaps it's phrased in a too specific way or too obscure  ...

Overall, as I said, I appreciate the effort. Look forward for the next revision :)

Akseluhr committed:
Fixing answers, questions and formatation

* Deleted answers that we need to discuss further
* Added and clarified questions and answers
* Fixed formatation

Akseluhr commented:
Hi again @bregman-arie. Thanks a lot for the interesting feedback. We decided to have further discussion among the two of us regarding the answers of the questions that were already there. Therefore, we decided to delete those answers for now.

On the contrary, we decided to focus on the questions we came up with regarding software supply chain and package management, and the feedback we received on it. 

Again, we are more than happy to address any issues with the latest commit if there are any. Thanks for the detailed answer and your time.

Best regards,
Aksel and Abdullah.

bregman-arie commented on exercises/security/README.md:
```diff
@@ -177,6 +229,9 @@ True. It is only used during the key exchange algorithm of symmetric encryption.
 
 <details>
 <summary>What is Hashing?</summary><br><b>
+Hashing is the process of converting a given key into another value. A hash function is used to generate the new value according to a mathematical algorithm. The result of a hash function is known as a hash value or simply, a hash.
```
Same comment

bregman-arie commented on exercises/security/README.md:
```diff
@@ -423,3 +417,145 @@ Practically this means that a single IP can server multiple web services/pages,
+#### Software Supply Chain & Security 
+
+<details>
+<summary>Briefly describe software supply chain. </summary><br><b>
```
Briefly describe what's a software supply chain

bregman-arie commented on exercises/security/README.md:
How is this question relevant to the topic? Also, AWS is not open source nor third party

Akseluhr replied:
Agree on AWS and the question's nature. We deleted the question and added an example to the question above. We think that having an example / few examples of what a component could be within the context of a software supply chain will contribute to clarifying this section for the reader.  Especially for those who are very new to DevOps / DevSecOps and possibly tech.

bregman-arie commented on exercises/security/README.md:
```diff
@@ -423,3 +478,133 @@ Practically this means that a single IP can server multiple web services/pages,
+</b></details>
+
+<details>
+<summary>What is a software supply chain attack? </summary><br><b>
```
Same comment

Akseluhr committed:
Removed and modified answers

- Removed answer about hashing
- Removed Q & A about a software supply chain attack (this one was basically a duplicate as well)
- Removed the commercial-question and answers
- Rephrased question about describing a software supply chain
- Fixed minor formatings

bregman-arie commented on exercises/security/README.md:
Missing quotation marks

bregman-arie commented on exercises/security/README.md:
Hmm, DepClean is one specific tool. Not sure it fits.

bregman-arie commented on exercises/security/README.md:
I would add a question of what is an actor in first place

Akseluhr replied:
"In first place" - do you mean in the beginning of the 'software supply chain' section or right before this question? 

We think that a question of what a threatening actor is should be enough for the reader to understand the question and answer, given the context (security) and the question's nature. Also, we believe that a question about what an actor is (generally speaking?) might not fit the context, but those are just our thoughts. In case you want a question about what an actor is, we'll of course add it (if so, please let us know in what context).

Akseluhr committed:
Merge branch 'bregman-arie:master' into master

Akseluhr committed:
Modified questions and answers

- Removed question about DepClean
- Added quoutes where needed
- Changed grammar of requested questions
- Changed formulation of a requested question

Akseluhr commented:
We are very happy with the feedback so fat, let us know if there is more to be addressed.

Akseluhr commented:
Hi @bregman-arie I hope u are doing well. We are just wondering whether you (or someone else) could have a check on the latest changes and potentially get it merged by Tuesday. All the best,

Aksel and abdullah

bregman-arie reviewed (approved):
Looks good. Thank you for the effort.

Akseluhr commented:
> 

Thanks for the feedback process, we really appreciate your efforts in this PR and the entire repo.

Have a good one!

Aksel & Abdullah
